dbms/src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -355,8 +355,20 @@ struct StorageDistributedDirectoryMonitor::Batch
             /// we must try to re-send exactly the same batches.
             /// So we save contents of the current batch into the current_batch_file_path file
             /// and truncate it afterwards if all went well.
-            WriteBufferFromFile out{parent.current_batch_file_path};
-            writeText(out);
+
+            /// Temporary file is required for atomicity.
+            String tmp_file{parent.current_batch_file_path + ".tmp"};
+
+            if (Poco::File{tmp_file}.exists()) {
```

Comment:
But this is a race condition. Truncate (the default) is better.

Reply:
This is simply to put a message into the log